JavaScript/JavaScript.sublime-syntax

```diff
@@ -164,12 +166,46 @@ contexts:
       push: jsdoc-block-tag
 
   jsdoc-block-tag:
+    - match: '{{jsdoc_block_tag_with_type}}'
+      scope: entity.other.attribute-name.documentation.js
+      set:
+        - match: '\s+(?=\{)'
+          set: jsdoc-type-expression
+        - match: ''
+          pop: 1
     - match: '{{jsdoc_block_tag}}'
       scope: entity.other.attribute-name.documentation.js
       pop: 1
     - match: (?=\S)|$
       pop: 1
 
+  jsdoc-type-expression:
+    - match: '\{'
+      scope: punctuation.definition.type.begin.js, meta.tag.jsdoc
+      set:
+        - match: '\}'
+          scope: punctuation.definition.type.end.js, meta.tag.jsdoc
+          pop: 1
+        - include: jsdoc-type-expression-body
+
+  jsdoc-type-expression-body:
+    - match: (?=\*/)
+      pop: 1
+    - match: '^\s*\*'
```

Comment:
A while ago, we started scoping the leading asterisk in `/**` style comments because some ST functionality was updated to handle it.
I'm not 100% sure this is the right scope but I'm on mobile and with limited time ATM, and figured a hint and vague review could be better than none...
```suggestion
    - match: '^\s*(\*)'
      captures:
        1: punctuation.definition.comment.js
```